### PR TITLE
[GX7315] Pair-up ticket with @jcruger for adding support for TTL

### PR DESCRIPTION
**Pull request by ywangii**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -202,6 +202,36 @@ Album.create({
 });
 ```
 
+We also support setting an optional expiration period called `TTL`(Time To Leave) for data expiration and removal. You can set up the `TTL` option either when creating the data entry or updating it. Once when you update the data entry, it will reset its `TTL`. 
+```js
```
Comment (jcrugzz) [inline README.md]: Specify that TTL is in seconds

File: test/integration/model.tests.js [hunk 2]
```diff
@@ -795,6 +802,137 @@ describe('Model', function () {
...
+
+          setTimeout(function () {
+            Foo.findOne({ fooId: four }, function (err, result) {
+              assume(err).is.falsey();
+              assume(result);
+              assume(result.fooId).equals(four);
+              done();
+            });
+          }, 3000);
+        });
```
Comment (jcrugzz) [inline test/integration/model.tests.js]: This should be longer. We should not find a result here

File: test/integration/model.tests.js [hunk 3]
```diff
@@ -795,6 +802,137 @@ describe('Model', function () {
...
+      });
+    });
+
+    it('should update a record in the database that will expire with given ttl', function (done) {
+      Foo.update({ entity: { fooId: five, something: 'foo' }, ttl: 2 }, function (err) {
+        assume(err).is.falsey();
+
+        Foo.findOne({ fooId: five }, function (error, result) {
```
Comment (jcrugzz) [inline test/integration/model.tests.js]: This initial findOne isn't necessary for what we are testing

File: test/integration/model.tests.js [hunk 4]
```diff
@@ -795,6 +802,137 @@ describe('Model', function () {
...
+          assume(result);
+          assume(result.fooId).equals(five);
+
+          setTimeout(function () {
+            Foo.findOne({ fooId: five }, function (er, res) {
+              assume(er).is.falsey();
+              assume(res).is.falsey();
+              done();
+            });
+          }, 5000);
```
Comment (jcrugzz) [inline test/integration/model.tests.js]: This can be shorter since the above ttl is 2 seconds

File: test/integration/model.tests.js [hunk 5]
```diff
@@ -795,6 +802,137 @@ describe('Model', function () {
...
+          assume(result.fooId).equals(six);
+
+          setTimeout(function () {
+            Foo.findOne({ fooId: six }, function (er, res) {
+              assume(er).is.falsey();
+              assume(res);
+              assume(res.fooId).equals(six);
+              done();
+            });
+          }, 2000);
```
Comment (jcrugzz) [inline test/integration/model.tests.js]: This should be higher so we can show it still expires after its ttl

File: test/integration/model.tests.js (hunk 4 above)
Comment (ywangii) [reply test/integration/model.tests.js]: updated

File: README.md (hunk 1 above)
Comment (ywangii) [reply README.md]: updated

File: test/integration/model.tests.js (hunk 3 above)
Comment (ywangii) [reply test/integration/model.tests.js]: removed

**Comment by jcrugzz** [comment]
It seems that cassandra is no longer working properly on travis which is unfortunate but the tests do pass.

**Pull request merged**